src/orange/surf/Involute.hh

```diff
@@ -0,0 +1,314 @@
+ * that extends to the origin. After which the angle of tangent point can be
+ * determined and used to calculate \em a via: /f[ a = angle - t_point /f]
+ *
+ * To calculate the intercept of two circles it is assumed that the two circles
+ * are \em xp axis. The coordinate along \em xp originating from the center of
+ * one of the circles is given by: /f[
+ * xp = (d^2 - r^2 + R^2)/(2d)
+ * /f]
+ * Where \em d is the distance between the center of the two circles, \em r is
+ * the radius of the displaced circle, and \em R is the radius of the circle at
+ * the orgin. Then the point \em yp can be obatined from: \f[
+ * yp = +- sqrt(R^2 - xp^2)
+ * \f]
+ * Then to convert ( \em xp , \em yp ) to ( \em x, \em y ) the following
+ * roation is applied:/f[ x = xp*cos(theta) - yp*sin(theta) y = yp*cos(theta) +
+ * xp*sin(theta) /f] Where \em theta is the angle between the \em xp axis and
+ * the \em x axis.
+ */
+CELER_FUNCTION SignedSense Involute::calc_sense(Real3 const& pos) const
+{
+    real_type const x = pos[0] - origin_[0];
+    real_type const y = pos[1] - origin_[1];
+
+    /*
+     * Calculate distance to origin and obtain t value for distance.
+     */
+    real_type const rxy2 = ipow<2>(x) + ipow<2>(y);
+    real_type const t_point2 = (rxy2 / ipow<2>(r_b_)) - 1;
+    real_type const t_point = std::sqrt(t_point2);
+
+    /*
+     * Check if Point is on involute.
+     */
+    real_type const angle = t_point + a_;
+    real_type const x_inv = r_b_
+                            * (std::cos(angle) + t_point * std::sin(angle));
+    real_type const y_inv = r_b_
+                            * (std::sin(angle) - t_point * std::cos(angle));
+
+    if (x == x_inv && y == y_inv)
+    {
+        return SignedSense::on;
+    }
+
+    /*
+     * Check if point is in defined bounds.
+     */
+    if (std::fabs(t_point2) < ipow<2>(tmin_))
+    {
+        return SignedSense::outside;
+    }
+    if (std::fabs(t_point2) > ipow<2>(tmax_))
+    {
+        return SignedSense::outside;
+    }
+
+    /*
+     * Check if point is in interval
+     */
+
+    // Describ
+
+    // Calculate tangents
+    real_type r = std::sqrt(ipow<2>(x) + ipow<2>(y));
+    real_type xp = (ipow<2>(r_b_)) / (2 * r);
+    real_type yp = std::sqrt(ipow<2>(r_b_) - ipow<2>(xp)) * sign_;
+    real_type theta = std::atan(y / x);
+    if (x < 0)
+    {
+        theta += pi;
+    }
+
+    Array<real_type, 2> point;
+    point[0] = xp * std::cos(theta) - yp * std::sin(theta);
+    point[1] = yp * std::cos(theta) + xp * std::cos(theta);
+
+    // Calculate angle of tangent
+    theta = std::acos(point[0]
+                      / std::sqrt(point[0] * point[0] + point[1] * point[1]));
+    if (point[1] < 0)
+    {
+        theta = (pi - theta) + pi;
+    }
+    real_type a1 = theta - t_point;
+    if (std::fabs(theta) < std::fabs(tmax_ + a_) && a1 >= a_)
+    {
+        return SignedSense::inside;
+    }
+
+    while (std::fabs(theta) < (std::fabs(tmax_ + a_)))
```

Comment:
tmax is already constrained in the contractors for the Involute in Involute.cc, and InvoluteSolver InvoluteSolver.hh

Reply:
I see:
```c++
CELER_EXPECT(std::fabs(tmax) < 2 * pi + std::fabs(tmin));
```
but no condition on tmin? Thanks for pointing those out. But my question is also why tmax/tmin can be negative? (Why are you checking the absolute value rather than the value?)  Is it possible to simplify the logic by processing the originally constructed values?

Reply:
That was a left er from when I was solving the negative and positive involutes separately.  I am adding `CELER_EXCEPT` for ensuring they are positive